caffe2/operators/accuracy_op.cc

```diff
@@ -14,21 +14,43 @@ bool AccuracyOp<float, CPUContext>::RunOnDevice() {
   DCHECK_EQ(label.dim32(0), N);
   Y->Resize(vector<TIndex>());
   const auto* Xdata = X.data<float>();
-  const auto* labeldata = label.data<int>();
+  const auto* labelData = label.data<int>();
+  const int top_k = top_k_;
```

Comment:
This code depends on top_k <= D. So I think we should add 

top_k = std::min(top_k, D)